Lib/multiprocessing/shared_memory.py

```diff
@@ -563,11 +714,53 @@ def get_server(self):
         'Better than monkeypatching for now; merge into Server ultimately'
         if self._state.value != State.INITIAL:
             if self._state.value == State.STARTED:
-                raise ProcessError("Already started server")
+                raise ProcessError("Already started SharedMemoryServer")
             elif self._state.value == State.SHUTDOWN:
-                raise ProcessError("Manager has shut down")
+                raise ProcessError("SharedMemoryManager has shut down")
             else:
                 raise ProcessError(
                     "Unknown state {!r}".format(self._state.value))
-        return _Server(self._registry, self._address,
-                       self._authkey, self._serializer)
+        return self._Server(self._registry, self._address,
+                            self._authkey, self._serializer)
+
+    def SharedMemory(self, size):
+        """Returns a new SharedMemory instance with the specified size in
+        bytes, to be tracked by the manager."""
+        conn = self._Client(self._address, authkey=self._authkey)
+        try:
+            sms = SharedMemory(None, flags=O_CREX, size=size)
+            try:
+                dispatch(conn, None, 'track_segment', (sms.name,))
+            except BaseException as e:
+                sms.unlink()
+                raise e
+        finally:
+            conn.close()
+        return sms
+
+    def ShareableList(self, sequence):
+        """Returns a new ShareableList instance populated with the values
+        from the input sequence, to be tracked by the manager."""
+        conn = self._Client(self._address, authkey=self._authkey)
+        try:
+            sl = ShareableList(sequence)
+            try:
+                dispatch(conn, None, 'track_segment', (sl.shm.name,))
+            except BaseException as e:
+                sl.shm.unlink()
+                raise e
+        finally:
+            conn.close()
+        return sl
+
+SharedMemoryManager.register('Barrier', threading.Barrier, BarrierProxy)
+SharedMemoryManager.register(
+    'BoundedSemaphore',
+    threading.BoundedSemaphore,
+    AcquirerProxy
+)
+SharedMemoryManager.register('Condition', threading.Condition, ConditionProxy)
+SharedMemoryManager.register('Event', threading.Event, EventProxy)
+SharedMemoryManager.register('Lock', threading.Lock, AcquirerProxy)
+SharedMemoryManager.register('RLock', threading.RLock, AcquirerProxy)
+SharedMemoryManager.register('Semaphore', threading.Semaphore, AcquirerProxy)
```

Comment:
Since `SharedMemoryManager` is a subclass of `SyncManager` aren't these already registered?

Reply:
It is no longer a subclass of SyncManager -- it is merely a subclass of BaseManager.  I changed this while writing the docs when I realized how confusing it was to try to explain why Value, Array, list, dict, etc. (distributed shared memory objects) were present on something devoted to non-distributed shared memory.